types/kafka-node/index.d.ts

```diff
@@ -1,99 +1,126 @@
-    close(cb: (error: any) => any): void;
-    sendOffsetCommitV2Request(group: string, generationId: number, memberId: string, commits: Array<OffsetCommitRequest>, cb: (error: any, data: any) => any): void;
+    close(callback?: () => void): void;
+    topicExists(topics: string[], callback: (err?: TopicsNotExistError | any) => any): void;
+    refreshMetadata(topics: string[], cb?: (error?: any) => any): void;
+    sendOffsetCommitV2Request(group: string, generationId: number, memberId: string, commits: OffsetCommitRequest[], cb: () => void): void;
+}
+
+export class KafkaClient extends Client {
+    constructor(options?: KafkaClientOptions);
+    connect(): void;
 }
 
-export declare class Producer {
-    constructor(client: Client, options?: any, customPartitioner?: any);
-    on(eventName: string, cb: () => any): void;
-    on(eventName: string, cb: (error: any) => any): void;
-    send(payloads: Array<ProduceRequest>, cb: (error: any, data: any) => any): void;
-    createTopics(topics: Array<string>, async: boolean, cb?: (error: any, data: any) => any): void;
-    close(cb: (error: any) => any): void;
+export class Producer {
+    constructor(client: Client, options?: ProducerOptions, customPartitioner?: any);
+    on(eventName: "ready", cb: () => any): void;
+    on(eventName: "error", cb: (error: any) => any): void;
+    send(payloads: ProduceRequest[], cb: (error: any, data: any) => any): void;
+    createTopics(topics: string[], async: true, cb: (error?: any, data?: any) => any): void;
+    createTopics(topics: string[], async: false, cb: () => any): void;
+    createTopics(topics: string[], cb: (error?: any, data?: any) => any): void;
+    close(): void;
 }
 
-export declare class HighLevelProducer {
-    constructor(client: Client, options?: any, customPartitioner?: any);
-    on(eventName: string, cb: () => any): void;
-    on(eventName: string, cb: (error: any) => any): void;
-    send(payloads: Array<ProduceRequest>, cb: (error: any, data: any) => any): void;
-    createTopics(topics: Array<string>, async: boolean, cb?: (error: any, data: any) => any): void;
-    close(cb: (error: any) => any): void;
+export class HighLevelProducer extends Producer {
 }
 
-export declare class Consumer {
-    constructor(client: Client, fetchRequests: Array<OffsetFetchRequest>, options: ConsumerOptions);
+export class Consumer {
+    constructor(client: Client, fetchRequests: Array<OffsetFetchRequest | string>, options: ConsumerOptions);
     client: Client;
-    on(eventName: string, cb: (message: string) => any): void;
-    on(eventName: string, cb: (error: any) => any): void;
-    addTopics(topics: Array<string>, cb: (error: any, added: boolean) => any): void;
-    addTopics(topics: Array<Topic>, cb: (error: any, added: boolean) => any, fromOffset: boolean): void;
-    removeTopics(topics: Array<string>, cb: (error: any, removed: boolean) => any): void;
+    on(eventName: "message", cb: (message: Message) => any): void;
```

Comment:
I've added a type to the message param in the callback. I'm not sure it is correct for the done event